pkg/operator/ceph/file/mds/mds.go

```diff
@@ -283,20 +283,14 @@ func prepareForDaemonUpgrade(
 	clusterName, fsName string,
 	timeout time.Duration,
 ) error {
+	//upgrade guide according to nautilus https://docs.ceph.com/docs/nautilus/cephfs/upgrading/#upgrading-the-mds-cluster
 	logger.Infof("preparing filesystem %s for daemon upgrade", fsName)
-	// * Beginning of noted section 1
-	// This section is necessary for upgrading to Mimic and to/past Luminous 12.2.3.
-	//   See more:  https://ceph.com/releases/v13-2-0-mimic-released/
-	//              http://docs.ceph.com/docs/mimic/cephfs/upgrading/
-	// As of Oct. 2018, this is only necessary for Luminous and Mimic.
-	if err := client.SetNumMDSRanks(context, cephVersion, clusterName, fsName, 1); err != nil {
+	if err := client.ReduceMDS(context, clusterName, fsName, 1); err != nil {
 		return errors.Wrapf(err, "Could not Prepare filesystem %s for daemon upgrade", fsName)
```

Comment:
While you are editing this, could you change this line? Error messages should not be capitalized.
```suggestion
		return errors.Wrapf(err, "could not prepare filesystem %q for daemon upgrade", fsName)
```